src/main/java/bookstore/repository/shoppingcart/ShoppingCartRepository.java

```diff
@@ -1,14 +1,12 @@
 package bookstore.repository.shoppingcart;
 
 import bookstore.model.ShoppingCart;
-import bookstore.model.User;
 import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
     @EntityGraph(attributePaths = {"items", "items.book"})
-    ShoppingCart getByUser(@Param("user") User user);
+    ShoppingCart getByUser_Email(String username);
```

Comment:
violates java code convention
```suggestion
    ShoppingCart getByUser_Email(String username);
```